File: Frontend/src/app/_service/vehicle-maintenance.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {environment} from "../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class VehicleMaintenanceService {

  maintenance;

  constructor(private http: HttpClient) { }

  //insert
  addMaintenance(maintenanceDetail): Observable<any> {
    return this.http.post<any>(environment.backend_url + "/maintenance/addVehicleMaintenance", maintenanceDetail);
  }

  chkVehicle(vehicleId): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/maintenance/chkVehicle/" + vehicleId);
  }

  //view
  getVehicleMaintenance(): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/maintenance/getVehicleMaintenance");
  }

  //update
  updateVehicleMaintenance(maintenanceDetail): Observable<any> {
    return this.http.put<any>(environment.backend_url + "/maintenance/updateVehicleMaintenance/" + maintenanceDetail.maintenanceID, maintenanceDetail);
  }

  //delete
  deleteVehicleMaintenance(vehicleMaintenanceID): Observable<any> {
    return this.http.delete<any>(environment.backend_url + "/maintenance/deleteVehicleMaintenance/" +vehicleMaintenanceID);
  }

  //get maintenance ID
  getMaintenanceById(vehicleAccidentID): Observable<any> {
    return this.http.get<any>(environment.backend_url + "/maintenance/getMaintenanceById/" + vehicleAccidentID);
  }


}

File: Backend/src/main/java/lk/fleet/service/VehicleMaintenanceService.java
package lk.fleet.service;

import lk.fleet.dto.VehicleAccidentDTO;
import lk.fleet.dto.VehicleMaintenanceDTO;
import lk.fleet.entity.VehicleMaintenance;

import java.util.List;

public interface VehicleMaintenanceService {

    VehicleMaintenanceDTO addMaintenance(VehicleMaintenance vehicleMaintenance);

    boolean chkVehicle(String vehicleId);

    VehicleMaintenanceDTO updateVehicleMaintenance(String maintenanceID, VehicleMaintenance vehicleMaintenance);

    boolean deleteVehicleMaintenance(String maintenanceID);

    List<VehicleMaintenanceDTO> getVehicleMaintenance();

    List<VehicleMaintenanceDTO> getMaintenanceById(String vehicleID);
}

